### ISSUE TITLE
Unable to switch to local when remote goes down

### ISSUE THREAD

**Issue by MikhailArkhipov** [Microsoft/RTVS#2599]
Connect to remote
Kill broker
Verify in REPL host disconnected
Try switching to local

Never switches

![image](https://cloud.githubusercontent.com/assets/12820357/19822721/da940f74-9d19-11e6-8b01-469279d069f3.png)

**Comment by MikhailArkhipov** [comment]
Related to #2570 ?

**Comment by MikhailArkhipov** [comment]
Hangs in
```
lockToken = await _connectArwl.WriterLockAsync(cancellationToken);
```

**Comment by MikhailArkhipov** [comment]
![image](https://cloud.githubusercontent.com/assets/12820357/19823209/4985c678-9d1c-11e6-9573-6211b054d655.png)

**Comment by AlexanderSher** [comment]
Fixed in #2570
